core/src/pages.rs

```diff
@@ -0,0 +1,467 @@
+    fn sub(&self, other: Self) -> Result<Self, PageError> {
+        self.sub_raw(other.raw())
+    }
+    /// Returns page which has number `page.raw() + 1`, with checks.
+    fn inc(&self) -> Result<Self, PageError> {
+        self.add_raw(1)
+    }
+    /// Returns page which has number `page.raw() - 1`, with checks.
+    fn dec(&self) -> Result<Self, PageError> {
+        self.sub_raw(1)
+    }
+    /// Aligns page zero byte and returns page which contains this byte.
+    /// Normally if `size % Self::size() == 0`,
+    /// then aligned byte is zero byte of the returned page.
+    fn align_down(&self, size: NonZeroU32) -> Self {
+        let size: u32 = size.into();
+        Self::from_offset((self.offset() / size) * size)
+    }
+    /// Returns page, which has zero byte offset == 0.
+    fn zero() -> Self {
+        unsafe { Self::new_unchecked(0) }
+    }
+    /// Returns iterator `self`..`self` + `count`.
+    fn iter_count(&self, count: Self) -> Result<PagesIter<Self>, PageError> {
+        self.add(count).map(|end| PagesIter { page: *self, end })
+    }
+    /// Returns iterator `self`..`end`.
+    fn iter_end(&self, end: Self) -> Result<PagesIter<Self>, PageError> {
+        if end.raw() >= self.raw() {
+            Ok(PagesIter { page: *self, end })
+        } else {
+            Err(PageError::WrongRange(self.raw(), end.raw()))
+        }
+    }
+    /// Returns iterator `self`..=`end`.
+    fn iter_end_inclusive(&self, end: Self) -> Result<PagesIterInclusive<Self>, PageError> {
+        if end.raw() >= self.raw() {
+            Ok(PagesIterInclusive {
+                page: Some(*self),
+                end,
+            })
+        } else {
+            Err(PageError::WrongRange(self.raw(), end.raw()))
+        }
+    }
+    /// Returns iterator `0`..=`self`
+    fn iter_from_zero_inclusive(&self) -> PagesIterInclusive<Self> {
+        PagesIterInclusive {
+            page: Some(Self::zero()),
+            end: *self,
+        }
+    }
+    /// Returns iterator `0`..`self`
+    fn iter_from_zero(&self) -> PagesIter<Self> {
+        PagesIter {
+            page: Self::zero(),
+            end: *self,
+        }
+    }
+    /// Returns iterator `self`..=`self`
+    fn iter_once(&self) -> PagesIterInclusive<Self> {
+        PagesIterInclusive {
+            page: Some(*self),
+            end: *self,
+        }
+    }
+    /// To another page iterator. For example: PAGE1 has size 4 and PAGE2 has size 2:
+    /// ````ignored
+    /// Memory is splitted into PAGE1:
+    /// [<====><====><====><====><====>]
+    ///  0     1     2     3     4
+    /// Memory splitted into PAGE2:
+    /// [<=><=><=><=><=><=><=><=><=><=>]
+    ///  0  1  2  3  4  5  6  7  8  9
+    /// Then PAGE1 with number 2 contains [4, 5] pages of PAGE2,
+    /// and we returns iterator over [4, 5] PAGE2.
+    /// ````
+    fn to_pages_iter<P: PageU32Size>(&self) -> PagesIterInclusive<P> {
```

Comment:
Hmm, seems like this comment is deprecated a little, could you update it please?